common/searchattribute/test_provider.go

```diff
@@ -112,7 +112,7 @@ func (t *TestMapper) GetFieldName(alias string, namespace string) (string, error
 	if namespace == "error-namespace" {
 		return "", serviceerror.NewInternal("mapper error")
 	} else if namespace == "test-namespace" || namespace == t.Namespace {
-		if alias == "pass-through" {
+		if alias == "pass-through" || alias == ScheduleID {
```

Comment:
`TestMapper` is actually not implicated for the tests I added, so I reverted this change entirely.

It's not as simple as returning WorkflowID if alias is ScheduleID. We need to test the flow when the SA exists and when it does not, and this custom mapper doesn't have that context. Rather than continuing to overload and complicate this mapper, I chose to override the mapper behavior on a per-test basis, as you can see in query_converter_test.go